libbuteosyncfw/profile/ProfileManager.cpp

```diff
@@ -739,6 +738,10 @@ QString ProfileManager::updateProfile(const Profile &aProfile)
     bool exists = d_ptr->profileExists(aProfile.name(),aProfile.type());
 
     QString profileId("");
+    if(d_ptr->save(aProfile)) {
```

Comment:
// We need to save before emit the signalProfileChanged, if this is the first
    // update the profile will only exists on disk after the save and any operation
    // using this profile triggered by the signal will fail.